code/modules/power/supermatter/supermatter.dm

```diff
@@ -495,17 +501,18 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 		if(damage > explosion_point)
 			countdown()
-
+	
+	last_complete_process = world.time
 	return 1
 
 /obj/machinery/power/supermatter_crystal/bullet_act(obj/item/projectile/Proj)
 	var/turf/L = loc
 	if(!istype(L))
 		return FALSE
-	if(!istype(Proj.firer, /obj/machinery/power/emitter) && power_changes)
+	if(!istype(Proj.firer, /obj/machinery/power/emitter))
 		investigate_log("has been hit by [Proj] fired by [key_name(Proj.firer)]", INVESTIGATE_ENGINES)
 	if(Proj.flag != "bullet")
-		if(power_changes) //This needs to be here I swear
+		if(is_power_processing()) //This needs to be here I swear //Okay bro, but I'm taking the other check because it definitely doesn't.
```

Comment:
I am mistaken, this will have a change since bullets are tied to a different subsystem than air. I feel like there is a cleaner way to do this considering that everything else should already be tied to air processing

Reply:
I'm not sure what you're asking me to do here. 

This check is just to prevent power surges between atmos processing ticks. If atmos pauses and emitters don't, power builds and builds and builds and finally atmos ticks again only to have it do so with a large amount of power built up. 

having this check here allows bullets to still come through, they just won't have any effect on the SM's power output.